modules/gapi/test/gapi_async_test.cpp

```diff
@@ -380,14 +380,117 @@ TYPED_TEST_P(cancel, basic){
     ASSERT_GT(canceled, 0u);
 }
 
+namespace {
+    GRunArgs deep_copy_out_args(const GRunArgsP& args ){
+        GRunArgs result; result.reserve(args.size());
+        for (auto&& arg : args){
+            switch (arg.index()){
+    #if !defined(GAPI_STANDALONE)
+                case GRunArgP::index_of<cv::Mat*>()                 :   result.emplace_back(*util::get<cv::Mat*>(arg));     break;
+                case GRunArgP::index_of<cv::Scalar*>()              :   result.emplace_back(*util::get<cv::Scalar*>(arg));  break;
+                case GRunArgP::index_of<cv::UMat*>()                :   result.emplace_back(*util::get<cv::UMat*>(arg));    break;
+    #endif // !defined(GAPI_STANDALONE)
+                case GRunArgP::index_of<cv::gapi::own::Mat*>()      :   result.emplace_back(*util::get<cv::gapi::own::Mat*>   (arg));   break;
+                case GRunArgP::index_of<cv::gapi::own::Scalar*>()   :   result.emplace_back(*util::get<cv::gapi::own::Scalar*>(arg));   break;
+                case GRunArgP::index_of<cv::detail::VectorRef>()    :   result.emplace_back(util::get<cv::detail::VectorRef>  (arg));   break;
+                default : ;
+            }
+        }
+        return result;
+    }
+
+    GRunArgsP args_p_from_args(GRunArgs& args){
+        GRunArgsP result; result.reserve(args.size());
+        for (auto&& arg : args){
+            switch (arg.index()){
+    #if !defined(GAPI_STANDALONE)
+                case GRunArg::index_of<cv::Mat>()                 :   result.emplace_back(&util::get<cv::Mat>(arg));     break;
+                case GRunArg::index_of<cv::Scalar>()              :   result.emplace_back(&util::get<cv::Scalar>(arg));  break;
+                case GRunArg::index_of<cv::UMat>()                :   result.emplace_back(&util::get<cv::UMat>(arg));    break;
+    #endif // !defined(GAPI_STANDALONE)
+                case GRunArg::index_of<cv::gapi::own::Mat>()      :   result.emplace_back(&util::get<cv::gapi::own::Mat>   (arg));   break;
+                case GRunArg::index_of<cv::gapi::own::Scalar>()   :   result.emplace_back(&util::get<cv::gapi::own::Scalar>(arg));   break;
+                case GRunArg::index_of<cv::detail::VectorRef>()   :   result.emplace_back(util::get<cv::detail::VectorRef>  (arg));   break;
+                default : ;
+            }
+        }
+        return result;
+    }
+}
+
 REGISTER_TYPED_TEST_CASE_P(cancel, basic);
 
+template<typename case_t>
+struct output_args_lifetime : ::testing::Test{
+    static constexpr const int num_of_requests = 20;
+};
+TYPED_TEST_CASE_P(output_args_lifetime);
+//There are intentionaly no actual checks (asserts and verify) in output_args_lifetime tests.
+//They are more of example use-cases nature than tests. (ASAN/valgrind can steal catch issues here)
```

Comment:
May be:
```
The test cases below describe user scenarios of handling output arguments lifetime when G-API Async API.
These scenarios can be considered examples rather than actual test cases.
```

Reply:
I would prefer to keep existing description

Reply:
okay